nibabel/cifti2/cifti2.py

```diff
@@ -0,0 +1,848 @@
+        if self.voxel_indices_ijk:
+            brain_model.append(self.voxel_indices_ijk._to_xml_element())
+        if self.vertex_indices:
+            brain_model.append(self.vertex_indices._to_xml_element())
+        return brain_model
+
+
+class Cifti2MatrixIndicesMap(object):
+    """Class for Matrix Indices Map
+
+    Provides a mapping between matrix indices and their interpretation.
+    """
+    # applies_to_matrix_dimension = list
+    # indices_map_to_data_type = str
+    # number_of_series_points = int
+    # series_exponent = int
+    # series_start = float
+    # series_step = float
+    # series_unit = str
+
+    def __init__(self, applies_to_matrix_dimension,
+                 indices_map_to_data_type,
+                 number_of_series_points=None,
+                 series_exponent=None,
+                 series_start=None,
+                 series_step=None,
+                 series_unit=None,
+                 brain_models=None,
+                 named_maps=None,
+                 parcels=None,
+                 surfaces=None,
+                 volume=None):
+        self.applies_to_matrix_dimension = applies_to_matrix_dimension
+        self.indices_map_to_data_type = indices_map_to_data_type
+        self.number_of_series_points = number_of_series_points
+        self.series_exponent = series_exponent
+        self.series_start = series_start
+        self.series_step = series_step
+        self.series_unit = series_unit
+        self.brain_models = brain_models if brain_models is not None else []
+        self.named_maps = named_maps if named_maps is not None else []
+        self.parcels = parcels if parcels is not None else []
+        self.surfaces = surfaces if surfaces is not None else []
+        self.volume = volume  # _value_or_make_klass(volume, Cifti2Volume)
+
+    def add_cifti_brain_model(self, brain_model):
+        """ Adds a brain model to the Cifti2MatrixIndicesMap
+
+        Parameters
+        ----------
+        brain_model : Cifti2BrainModel
+        """
+        if not isinstance(brain_model, Cifti2BrainModel):
+            raise TypeError("Not a valid Cifti2BrainModel instance")
+        self.brain_models.append(brain_model)
+
+    def remove_cifti_brain_model(self, ith):
+        """ Removes the ith brain model element from the Cifti2MatrixIndicesMap """
+        self.brain_models.pop(ith)
+
+    def add_cifti_named_map(self, named_map):
+        """ Adds a named map to the Cifti2MatrixIndicesMap
+
+        Parameters
+        ----------
+        named_map : Cifti2MatrixIndicesMap
+        """
+        if isinstance(named_map, Cifti2MatrixIndicesMap):
+            raise TypeError("Not a valid Cifti2MatrixIndicesMap instance")
+        self.named_maps.append(named_map)
+
+    def remove_cifti_named_map(self, ith):
+        """ Removes the ith named_map element from the Cifti2MatrixIndicesMap """
+        self.named_maps.pop(ith)
+
+    def add_cifti_parcel(self, parcel):
```

Comment:
If you felt fancy, I bet you could make a nice decorator to do this repetitive logic for the `append / pop` methods.